docs/content/latest/api/ycql/batch.md

```diff
@@ -0,0 +1,173 @@
+the higher the latency for the entire batch. Although the latency for the entire batch of operations is higher than the latency of any single operation, 
+the throughput of the batch of operations is much higher.
+
+## Example in JAVA
+
+In order to perform a batch insert operation in Java, first create a BatchStatement object. 
+Next add the desired number of prepared and bound insert statements to it. 
+Finally, execute the batch object. This is shown below.
+
+```java
+// Create a batch statement object.
+BatchStatement batch = new BatchStatement();
+
+// Create a prepared statement object to add to the batch.
+PreparedStatement insert = client.prepare("INSERT INTO table (k, v) VALUES (?, ?);");
+
+// Bind values to the prepared statement and add them to the batch.
+for (...) {
+  batch.add(insert.bind( ... <values for bind variables> ... ));
+}
+
+// Execute the batch operation.
+ResultSet resultSet = client.execute(batch);
+
+```
+## Example in Python using RETURNS AS STATUS
+
+An example using Python client and RETURNS AS STATUS clause:
+
+```cassandra
+ycqlsh> create keyspace if not exists yb_demo;
+ycqlsh> CREATE TABLE if not exists yb_demo.test_rs_batch(h int, r bigint, v1 int, v2 varchar, primary key (h, r));
+ycqlsh> INSERT INTO yb_demo.test_rs_batch(h,r,v1,v2) VALUES (1,1,1,'a');
+ycqlsh> INSERT INTO yb_demo.test_rs_batch(h,r,v2) VALUES (3,3,'b');
+ycqlsh> select * from yb_demo.test_rs_batch;
+
+ h | r | v1   | v2
+---+---+------+----
+ 1 | 1 |    1 |  a
+ 3 | 3 | null |  b
+
+(2 rows)
+```
+
+Getting status from DML operations in Python:
+
+```python
+from cassandra.cluster import Cluster, BatchStatement
+
+# Create a cluster and a session
+cluster = Cluster(['127.0.0.1'])
+session = cluster.connect()
+
+# Create a batch statement object.
+b = BatchStatement()
+
+# Add multiple queries
+b.add(f"INSERT INTO test_rs_batch(h, r, v1, v2) VALUES (1, 1, 1 ,'a') RETURNS STATUS AS ROW;")
+b.add(f"UPDATE test_rs_batch SET v2='z' WHERE h=3 AND r=3 IF v2='z'  RETURNS STATUS AS ROW;")
+b.add(f"DELETE FROM test_rs_batch WHERE h=2 AND r=2 IF EXISTS RETURNS STATUS AS ROW;")
+
+# Execute the batch operation.
+result = session.execute(b, trace=True)
+
+# Print status for each DML operation
+for row in result:
+    print(row)
+```
+
+The output generated is:
+
+```python
+Row(applied=True, message=None, h=None, r=None, v1=None, v2=None)
+Row(applied=False, message=None, h=3, r=3, v1=None, v2='b')
+Row(applied=False, message=None, h=None, r=None, v1=None, v2=None)
+```
+
+## Row Status
+
+When executing a batch in YCQL, the protocol just allows returning one error/return status.
+
+So, if one statement fails with an error or, for conditional DMLs, some are not applied because of failing the IF condition, 
+the driver/application cannot accurately identify the relevant statement(s), it will just receive one general error/return-status for the whole batch.
```

Comment:
```suggestion
If one statement fails with an error or for conditional DMLs, some are not applied because of failing the IF condition, the driver or application cannot accurately identify the relevant statements, it will just receive one general error or return-status for the batch.
```